src/core/worker.js

```diff
@@ -193,6 +193,14 @@ class WorkerMessageHandler {
         pdfManager.ensureDoc("fingerprint"),
         pdfManager.ensureDoc("isPureXfa"),
       ]);
+
+      if (isPureXfa) {
+        const task = new WorkerTask("Load fonts for Xfa");
+        startWorkerTask(task);
+        await pdfManager
+          .loadXfaFonts(handler, task)
+          .finally(() => finishWorkerTask(task));
```

Comment:
If the `loadXfaFonts` method would ever *reject*, I believe that this will break all PDF document loading (since `finally` doesn't stop error-propagation); hence please do:
```suggestion
        await pdfManager
          .loadXfaFonts(handler, task)
          .catch(reason => {
            // Ignore errors, to allow the document to load.
          })
          .then(() => finishWorkerTask(task));
```